### DISCUSSION TITLE
Allow for API key authentication in POST body

### DISCUSSION THREAD

**Discussion by sebastianhutter** [grafana/grafana-infinity-datasource#585] [Ideas]
I am using the Infinity plugin to query the [Mandrill API](https://mailchimp.com/developer/transactional/api/) to retrieve usage statistics. The API only accepts POST requests and requires the API key to be part of the POST body (`key` field). 

The infinity plugin currently does not support this kind of authentication which means I need to add the key to every panel.
![image](https://user-images.githubusercontent.com/2313134/233316786-559ba34c-b740-41df-a310-54db989c1cce.png)

It would be great if the ["API Key Authentication"](https://sriramajeyam.com/grafana-infinity-datasource/wiki/authentication/) could be extended to support defining API keys in the request body.

**Comment by yesoreyeram** [comment]
It is is bit harder than it seems but doable. ( not in my roadmap.. but PRs accepted. )

Not everyone will use the body of json type.  Also not every api will accept json as the straight key-value pair/object. some json body may have nested fields.  

Someone else might use form-data as body..
